### DISCUSSION TITLE
.clinerules file

### DISCUSSION THREAD

**Discussion by jonit-dev** [cline/cline#622] [Feature Requests]
A per project custom instruction file, that cline uses to change its behavior based on the codebase

**Comment by lloydchang** [comment]
as an alternative, you can try a fork https://github.com/RooVetGit/Roo-Cline/commit/a6bfbf8bcb4b27ae02a8ae35c4b0fdd5a5417416#diff-ec5bfb97fa984921dcb420c3fcf9747854c3302ac5ec437f1fe604ee68fc4e7aR287
> `const ruleFiles = ['.clinerules', '.cursorrules']`

and its `.vsix` at https://marketplace.visualstudio.com/items?itemName=RooVeterinaryInc.roo-cline

from https://discord.com/channels/1275535550845292637/1275535550845292640/1312113711842066524

@mrubens wrote:
> In the fork I've been working on I have Cline reading .cursorrules and .clinerules from the working directory and interpolating them into the system prompt if present (at least in our use case most of the things we tell Cursor are also relevant for Cline). 

related https://github.com/cline/cline/discussions/361 https://github.com/cline/cline/discussions/408 https://github.com/cline/cline/discussions/622

**Reply by mxro** [reply]
I think this is implemented now with V3.0 - but I cannot seem to find a reference what kind of content is supported in the file? I assume it's just a plain text prompt?

**Reply by mrubens** [reply]
At least in Roo it just gets appended to the bottom of the system prompt

**Comment by saoudrizwan** [comment]
Added in Cline v3!

You can now specify custom instructions for your project by adding a root-level .clinerules file. This allows you to define project-specific behaviors and guidelines for Cline–perfect for setting conventions, pointing to important documentation, or providing context about your project's architecture.
